homeworkw4d2.rb

```diff
@@ -0,0 +1,176 @@
+      last_name: 'Almohsen',
+  }
+]
+
+# upper_case_full_names = []
+
+upper_cased_names = students.map { |name| name.upcase }
+
+# 2. Find the first order for each user
+
+users = [
+  {
+      name: 'Hanan',
+      orders: [
+          {
+              description: 'a bike'
+          }
+      ]
+  },
+  {
+      name: 'Abdulrahman',
+      orders: [
+          {
+              description: 'bees'
+          },
+          {
+              description: 'fishing rod'
+          }
+      ]
+  },
+  {
+      name: 'Turki',
+      orders: [
+          {
+              description: 'a MacBook'
+          },
+          {
+              description: 'The West Wing DVDs'
+          },
+          {
+              description: 'headphones'
+          },
+          {
+              description: 'a kitten'
+          }
+      ]
+  }
+]
+
+
+description = users.map { |d| d["description"] }
+
+
+
+
+# 3. Find the average amount spent on coffee, per transaction, for each person
+
+people = [
+  {
+      name: 'Ahlam',
+      transactions: [
+          {
+              type: 'COFFEE',
+              amount: 7.43
+          },
+          {
+              type: 'TACOS',
+              amount: 14.65
+          },
+          {
+              type: 'COFFEE',
+              amount: 4.43
+          }
+      ]
+  },
+  {
+      name: 'Sulaiman',
+      transactions: [
+          {
+              type: 'BIKES',
+              amount: 800.00
+          },
+          {
+              type: 'TACOS',
+              amount: 14.65
+          },
+          {
+              type: 'COFFEE',
+              amount: 4.43
+          }
+      ]
+  },
+  {
+      name: 'Norah',
+      transactions: [
+          {
+              type: 'COFFEE',
+              amount: 7.43
+          },
+          {
+              type: 'COFFEE',
+              amount: 100.00
+          },
+          {
+              type: 'COFFEE',
+              amount: 4.43
+          }
+      ]
+  }
+]
+
+
+amount = people.map { |o| o["amount"] }
+amount.inject{ |sum, el| sum + el }.to_f / arr.size
+
+
+# 4. Find the most expensive product for each store, with the store name:
+
+stores = [
+  {
+      store_name: 'Jarir',
+      products: [
+          {
+              description: 'Titanium',
+              price: 9384.33
+          },
+          {
+              description: 'Gold',
+              price: 345.54
+          }
+      ]
+  },
+  {
+      store_name: 'Tamimi',
+      products: [
+          {
+              description: 'Silver',
+              price: 654.44
+          },
+          {
+              description: 'Ruby',
+              price: 323.43
+          }
+      ]
+  },
+  {
+      store_name: 'Souq',
+      products: [
+          {
+              description: 'Opal',
+              price: 345.43
+          },
+          {
+              description: 'Sapphire',
+              price: 899.33
+          }
+      ]
+  }
+]
+
+def max_price(stores)
+    [stores.max]
+  end
```

Comment:
Why are you using brackets?
The .max method wouldn't work on an array of hashes